test_rclcpp/test/test_timer.cpp

```diff
@@ -108,16 +109,15 @@ TEST(CLASSNAME(test_time, RMW_IMPLEMENTATION), timer_during_wait) {
   ASSERT_EQ(0, counter);
 
   auto spinner =
-    [&executor, &node]() -> void
+    [&executor]() -> void
     {
       printf("spin() until shutdown\n");
-      executor.add_node(node, false);
       executor.spin();
     };
 
   auto thread = std::thread(spinner);
-  printf("sleeping for 4 periods\n");
-  std::this_thread::sleep_for(4 * period);
+  printf("sleeping for 3.5 more periods\n");
+  std::this_thread::sleep_for(std::chrono::duration_cast<std::chrono::nanoseconds>(3.5 * period));
```

Comment:
Wow, I was not aware that this is the current behavior of a timer. We should definitely change it to fire at `now() + period` for the first time.

Also I don't see what the `spin()` call should have to do with it. When I create the timer at `t1` I expect the first callback at `t1 + period` independent on when `spin()` was invoked (obviously only when some executor is handling it, otherwise the callback gets delayed until it is being handled).